content/docs/1.3.3/volumes-and-nodes/maintenance.md

```diff
@@ -71,14 +64,37 @@ To remove a node:
 1. Evict all the replicas on the node.
 1. Detach all the volumes on the node.
 
-    If the node has been drained, all the workloads should be migrated to another node already.
+   If the node has been drained, all the workloads should be migrated to another node already.
 
-    If there are any other volumes remaining attached, detach them before continuing.
+   If there are any other volumes remaining attached, detach them before continuing.
 
 1. Remove the node from Longhorn using the `Delete` in the `Node` tab.
 
-    Or, remove the node from Kubernetes, using:
+   Or, remove the node from Kubernetes, using:
 
         kubectl delete node <node-name>
 
 1. Longhorn will automatically remove the node from the cluster.
+
+## Upgrading Kubernetes
+
+### In-place Upgrade
+In-place upgrade is upgrading method in which nodes are upgraded without being removed from the cluster.
+Some example solutions that use this upgrade methods are [k3s automated upgrades](https://docs.k3s.io/upgrades/automated), [Rancher's Kubernetes upgrade guide](https://rancher.com/docs/rancher/v2.x/en/cluster-admin/upgrading-kubernetes/#upgrading-the-kubernetes-version),
+[Kubeadm upgrade](https://kubernetes.io/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade/), etc...
+
+With the assumption that node and disks are not being deleted/removed, the recommended upgrading guide is:
+1. You should cordon and drain a node before upgrading Kubernetes components on a node.
+   Draining instruction is similar to the drain instruction at [Updating the Node OS or Container Runtime](#updating-the-node-os-or-container-runtime)
+2. The drain `--timeout` should be big enough so that replica rebuildings on healthy node can finish.
+   The more Longhorn replicas you have on the draining node, the more time it takes for the Longhorn replicas to be rebuilt on other healthy nodes.
+   We recommending you to test and select a big enough value or set it to 0 (aka never timeout).
+3. The number of nodes doing upgrade at a time should be smaller than the number of Longhorn replicas for each volume.
+   This is so that a running Longhorn volume has at least one healthy replica running at a time.
+4. Set the setting [Node Drain Policy](../../references/settings#node-drain-policy) to `allow-if-replica-is-stopped` so that the drain is not blocked by the last healthy replica of a detached volume.
```

Comment:
We can suggest users have an appropriate value for setting `replica-replenishment-wait-interval`, which can cooperate with this policy and avoid some unnecessary replica rebuilding.

Reply:
FYI: Here is a request ticket that modify the default value of this setting to 1800:
https://github.com/harvester/harvester/issues/3671

Reply:
Sorry, What do you think would be the appropriate value for `replica-replenishment-wait-interval` in this draining case? 

Usually, node will be uncordoned immediately after finishing draining so it shouldn't be more than 10 minutes

Reply:
I don't have such a number in my mind since I am not experienced in the upgrade part.
But we can ask users to check and modify it based on their env/estimation here.

Reply:
From my testing, the node will be uncordoned immediately after the replica instance manager is successfully drained. So I think the default 10 mins should be more than enough. Do you think we should still have the note?

Reply:
If the default value is good enough, then we don't need to emphasize another value here.